tests/listusers/listusers_test.go

```diff
@@ -0,0 +1,274 @@
+package listusers
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"testing"
+
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+	parser "github.com/openfga/language/pkg/go/transformer"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/goleak"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zaptest/observer"
+	"google.golang.org/grpc"
+
+	"github.com/openfga/openfga/cmd/run"
+	"github.com/openfga/openfga/internal/mocks"
+	"github.com/openfga/openfga/internal/server/config"
+	"github.com/openfga/openfga/pkg/logger"
+	"github.com/openfga/openfga/pkg/testutils"
+	"github.com/openfga/openfga/pkg/typesystem"
+	"github.com/openfga/openfga/tests"
+)
+
+func TestListUsersMemory(t *testing.T) {
+	testRunAll(t, "memory")
+}
+
+func TestListUsersPostgres(t *testing.T) {
+	testRunAll(t, "postgres")
+}
+
+func TestListUsersMySQL(t *testing.T) {
+	testRunAll(t, "mysql")
+}
+
+func testRunAll(t *testing.T, engine string) {
+	t.Cleanup(func() {
+		goleak.VerifyNone(t)
+	})
+	cfg := config.MustDefaultConfig()
+	cfg.Log.Level = "error"
+	cfg.Datastore.Engine = engine
+	cfg.Experimentals = []string{"enable-list-users"}
+
+	tests.StartServer(t, cfg)
+
+	conn := testutils.CreateGrpcConnection(t, cfg.GRPC.Addr)
+
+	RunAllTests(t, openfgav1.NewOpenFGAServiceClient(conn))
+}
+
+func TestListUsersLogs(t *testing.T) {
```

Comment:
```suggestion
func TestListUsersTelemetry(t *testing.T) {
```

Since we're testing more than Logs - we're also testing exported metrics.